src/main/java/internet/shop/dao/jdbc/ShoppingCartDaoJdbcImpl.java

```diff
@@ -0,0 +1,162 @@
+package internet.shop.dao.jdbc;
+
+import internet.shop.dao.ShoppingCartDao;
+import internet.shop.exceptions.DataProcessingException;
+import internet.shop.lib.Dao;
+import internet.shop.model.Product;
+import internet.shop.model.ShoppingCart;
+import internet.shop.util.ConnectionUtil;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+@Dao
+public class ShoppingCartDaoJdbcImpl implements ShoppingCartDao {
+    @Override
+    public Optional<ShoppingCart> getByUserId(Long id) {
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement
+                    = connection.prepareStatement(
+                    "SELECT * FROM shopping_carts WHERE user_id = ?;");
+            statement.setLong(1, id);
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                ShoppingCart shoppingCart = getCartFromResultSet(resultSet);
+                for (Product product : getProductsFromCart(shoppingCart.getId())) {
+                    shoppingCart.addProduct(product);
+                }
+                return Optional.of(shoppingCart);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get shopping cart with user id: " + id, e);
+        }
+        return Optional.empty();
+    }
+
+    @Override
+    public ShoppingCart create(ShoppingCart cart) {
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement
+                    = connection.prepareStatement(
+                    "INSERT INTO shopping_carts(shopping_cart_id,user_id) VALUES (?, ?);");
+            statement.setLong(1, cart.getId());
+            statement.setLong(2, cart.getUserId());
+            statement.executeUpdate();
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't create shopping cart", e);
+        }
+        return cart;
+    }
+
+    @Override
+    public Optional<ShoppingCart> get(Long id) {
```

Comment:
I think you should also add products to the cart in this method.

Reply:
Done :)